### DISCUSSION TITLE
[FEATURE] Display full content of posts on homepage

### DISCUSSION THREAD

**Discussion by dbrg77** [HEIGE-PCloud/DoIt#507] [General]
### Describe the feature you want 描述你的功能需求

Hi,

Thanks for creating the awesome theme. I really like it. Currently, the default behaviour of `DoIt` is to display the summary of the post on homepage.

Is there a way of showing the full content for the post on homepage? I tried to modify a few things in `layouts` without success. I'm quite new to `Hugo`.

Thank you.

Regards,
Xi

**Comment by WingsZeng** [comment]
Maybe you can set `summaryLength` in the site configuration file to a large enough number?

Related documentation:
- https://gohugo.io/content-management/summaries/#automatic-summary-splitting

**Comment by dbrg77** [comment]
Thanks for the suggestion. I could do that, but the full content will be shown as plain text, not rendered in markdown.

The motivation here is to be able to immediately read the post in its full, rendered format, without having to click on the `Read more`. In other themes, I can locate the HTML file that control this behaviour and simply change the `.Summary` to `.Content`.

`DoIt` is just a bit too complicated for me to mess around 😂

**Comment by HEIGE-PCloud** [comment]
You are not recommended to do this, and DoIt probably will not add support for this feature either.

If you really want to mess around, you can change

https://github.com/HEIGE-PCloud/DoIt/blob/e177729e99631cff6cd79470e9440e8c7ca60517/layouts/_default/summary.html#L85
 
to `{{- with .Content -}}`

and remove this line

https://github.com/HEIGE-PCloud/DoIt/blob/e177729e99631cff6cd79470e9440e8c7ca60517/assets/css/_page/_home.scss#L129

will do the trick.

**Comment by dbrg77** [comment]
Thank you for the tip.

The outcome is not very satisfying by doing that. If it is not recommended, I will stick to the summary for now.

Thanks again.
